### DISCUSSION TITLE
How to get current page url (hostname) in client component ssr?

### DISCUSSION THREAD

**Discussion by siamahnaf** [vercel/next.js#61830] [Help]
### Summary

Hello, In Nextjs Server Component, there is a easy way to get host using headers.

import { headers } from "next/headers";

but in client component, I need to know the current hostname in ssr. How can I do that-

If your suggestion is use "window.location.hostname", then the whole page is not rendering in ssr, I need to use then useEffect, then, I am getting the hostname in csr. I need to know the hostname when suspense query that means, in ssr? How can I get hostname?

One possible solution is-
getting the hostname form a server component and pass that as props in the client component, in that case If my component tree will very complex, how can I manage it. Just give me some idea..!!

**Comment by DevDaniloFerrari** [comment]
I think that you can do something like that:

```js
'use client'
import { useEffect, useState } from 'react';

function MyComponent() {
  const [currentUrl, setCurrentUrl] = useState('');

  useEffect(() => {
    // Check if the code is running on the client side
    if (process) {
      // Access the current page URL using window.location
      setCurrentUrl(window.location.href);
    }
  }, []);

  return (
    <div>
      <p>Current URL: {currentUrl}</p>
      {/* Other components */}
    </div>
  );
}

export default MyComponent;
```

**Comment by siamahnaf** [comment]
I already try it, it is not ssr I think. I need to know hostname in ssr in client component-

[Video_2024_02_09-1.webm](https://github.com/vercel/next.js/assets/78312010/7768467d-ece0-4a50-9df2-35a5072371d3)

**Comment by makibat** [comment]
You can get the value of the host requested by the client using the `Headers` object, which can be easily accessed using the `headers` function from `next/headers`

``` js
import { headers } from 'next/headers';

const Page = () => {
  const headersList = headers();
  const hostname = headersList.get('x-forwarded-host');

  return (
    <div>host: {hostname}</div>
  );
};

export default Page;
```

Hope that helps!

**Reply by siamahnaf** [reply]
It is server components. I said I need it in client components

**Reply by makibat** [reply]
You can pass it as a prop to the client component.

Edit: or you can use the `useContext` hook if it's more intuitive for your file structure.

**Reply by abaracedo** [reply]
I think that's the unique way to know in a Client Component the host... :shrug: 

I think you need to give us more information about that you want to do and why is not valid to pass the host as prop or using a `useEffect` to get it...
